### DISCUSSION TITLE
Can I run `FlywayMIgrator` independently?

### DISCUSSION THREAD

**Discussion by TaKO8Ki** [phax/phoss-smp#253] [Q&A]
Hi. Thank you for your great work. Your SMP implementation is extremely helpful to me.

I have one question. Currently, phoss-smp handles migration using Flyway during server startup. Is it possible to execute it separately? In detail, can I run [FlywayMIgrator](https://github.com/phax/phoss-smp/blob/7d42e52c282a5fdc8ae505e8d70f43caf5d745ce/phoss-smp-backend-sql/src/main/java/com/helger/phoss/smp/backend/sql/mgr/FlywayMigrator.java#L56) independently? I'd like to run migrations beforehand and then deploy the SMP application. If this is possible, I plan to set `smp.flyway.enabled` to false on the application side and execute migrations separately.

**Comment by phax** [comment]
Hi @TaKO8Ki, I think I understand your request.
Currently it is not possible to "just" run the FlywayMigrator via an external trigger.

What kind of trigger are you thinking about? Some kind of "main application" based on a separate Docker image is what comes to my mind.
Because you need
* the latest code base
* the configuration file for the DB properties
and of course
* the trigger should not be accessible from the outside...

Any good proposals on what trigger to use are welcome

**Reply by TaKO8Ki** [reply]
I believe it would be a good idea to enable migration through the CLI. For instance, in the web framework, Ruby on Rails, you can start the server with `rails server` and perform migrations with `rails db:migrate` through the same CLI. Of course these actions can't be executed externally. As far as I know, Flyway can also be used as a command-line tool, which might be a suitable approach.

ref: https://documentation.red-gate.com/fd/quickstart-command-line-184127576.html
